portal/academy/views.py

```diff
@@ -200,14 +227,19 @@ def get(self, request, *args, **kwargs):
             user_data["submission_date"] = grade.created
             object_list.append(user_data)
 
-        context = self.get_context_data(
-            object_list=object_list,
-            spc_list=spc_list,
-            unit_list=unit_list,
-            max_score=max_score,
-            workspace_url=settings.SLACK_WORKSPACE,
-        )
-        return self.render_to_response(context)
+        if "download" in kwargs and kwargs["download"] == "csv":
+            response = HttpResponse(csvdata(spc_list, unit_list, object_list), content_type="text/csv")
+            response["Content-Disposition"] = "attachment; filename=test.csv"
```

Comment:
I would suggest a more descriptive filename.

Reply:
Changed to `student-grades.csv`